src/en/dynasty/src/eu/kanade/tachiyomi/extension/en/dynasty/DynastyDoujins.kt

```diff
@@ -64,7 +74,38 @@ class DynastyDoujins : DynastyScans() {
                 )
             }
         }
+        chapters.addAll(doujinChapterParse(document))
+
+        var hasNextPage = popularMangaNextPageSelector().let { selector ->
+            document.select(selector).first()
+        } != null
+
+        while (hasNextPage) {
+            page += 1
+            val doujinURL = document.location() + "?page=$page"
 
+            val newRequest = GET(doujinURL, headers)
+            val newResponse = client.newCall(newRequest).execute()
+
+            if (!newResponse.isSuccessful) {
+                retry += 1
+                if (retry > maxRetry) { break }
+                else {
+                    /*
+                    TODO: Toast to notify max retries exceeded
+                          and chapter parsing aborted
+                     */
+                    return chapters
+                }
+            }
```

Comment:
```kotlin
if (!newResponse.isSuccessful) {
    retry += 1
    if (retry > maxRetry) { break } // THIS is where max retries exceeded
    else { // THIS is where you are gonna retry
        /*
        TODO: Toast to notify max retries exceeded
              and chapter parsing aborted
         */
        return chapters
    }
}
```

I think you got it wrong. What I meant was, when you're gonna retry, you continue and hit page++ and actually "retry" on the next page. I think you should make a function to handle retrying, or just yeet the retrying logic and just toast and abort.

Reply:
We'll just do the KISS option for now and remove the retry logic, especially since I don't have a better plan in mind at the moment.

As a side note, I do not know if [advancing to an unintended page count](https://dynasty-scans.com/doujins/touhou_project?page=666&view=chapters) produces any unintended behavior.

Reply:
I mean, if you get an HTTP 404 when requesting page 3, on the next loop you'll request page 4. Normal retrying logic should be requesting page 3. If skipping page 3 is correct, this is not "retrying", but "skipping error pages" - where the max retries count doesn't make sense.

Reply:
I see what you're trying to get at now. Here's what I was going for. I'm not worried about 404 errors since the URL always produces a valid page. Rather, Dynasty has a tendency to temporarily return 50X errors for a few seconds up to minutes at a time. I wanted to retry the current page in this case until either it resolves or we run out of retry attempts. If we run out, try to march through the rest of the pages without continuing to retry. I think the way I implemented it before would have caused unintended behavior if the page it 50X's on is actually the last page, and with no kind of back off it would hammer the site potentially causing additional problems on their end.